### PR TITLE
cks: Get caller user keys if cluster belongs to project

### PR DESCRIPTION
**Pull request by davidjumani**
### Description

Fixes https://github.com/apache/cloudstack/issues/6344
Passes the caller's API keys if the cluster is part of a project

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?

autoscaler logs :

```
I0516 08:57:51.486442       1 static_autoscaler.go:235] Starting main loop
I0516 08:57:51.487030       1 client.go:169] NewAPIRequest API request URL:http://10.0.32.136:8080/client/api?apiKey=***&command=listKubernetesClusters&id=a5e617a3-6d95-410c-9d3a-306274ee8b11&response=json&signature=***
I0516 08:57:51.515655       1 client.go:175] NewAPIRequest response status code:200
I0516 08:57:51.518415       1 cloudstack_manager.go:88] Got cluster : &{a5e617a3-6d95-410c-9d3a-306274ee8b11 cluster 1 2 1 1 [0xc000716f90 0xc000716fc0] map[cluster-control-180cc02282f:0xc000716f90 cluster-node-180cc025111:0xc000716fc0]}

```

### CODE REVIEW

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan test

**Review by rohityadavcloud — APPROVED** [review]
LGTM, didn't test it

**Review by weizhouapache — APPROVED** [review]
code lgtm

**Comment by weizhouapache** [comment]
cc @nvazquez

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImpl.java [hunk 1]
```diff
@@ -1147,6 +1147,14 @@ public boolean startKubernetesCluster(long kubernetesClusterId, boolean onCreate
         }
     }
 
+    private Account getOwnerOrCaller(KubernetesClusterVO kubernetesCluster) {
+        Account owner = accountService.getActiveAccountById(kubernetesCluster.getAccountId());
+        if (owner.getType() == Account.Type.PROJECT) {
```
Comment (sureshanaparti) [inline plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImpl.java:1152]:
```suggestion
        if (owner != null && owner.getType() == Account.Type.PROJECT) {
```

Comment (shwstppr) [inline plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImpl.java:1152]:
```suggestion
        if (owner == null || owner.getType() == Account.Type.PROJECT) {
```
@davidjumani maybe this and refactor `getServiceUserKeys`

**Comment by davidjumani** [comment]
@blueorangutan package

**Review by shwstppr — APPROVED** [review]
Code LGTM

**Comment by shwstppr** [comment]
@blueorangutan package

**Review by sureshanaparti — APPROVED** [review]
code LGTM

**Comment by weizhouapache** [comment]
@blueorangutan test

**Review by nvazquez — APPROVED** [review]

**Pull request merged**